tests/qeidentity/host/qeidentifyinfo.cpp

```diff
@@ -139,12 +163,232 @@ void run_qe_identity_test_cases(oe_enclave_t* enclave)
+    platform_tcb_level.isvsvn[0] = 1;
+    OE_TEST(
+        test_verify_qe_identity_info(
+            enclave,
+            &ecall_result,
+            (const char*)&positive_qe_id_info[0],
+            &platform_tcb_level,
+            &parsed_info) == OE_OK);
+    OE_TEST(ecall_result == OE_TCB_LEVEL_INVALID);
+    OE_TEST(parsed_info.id == QE_IDENTITY_ID_QE);
+    OE_TEST(parsed_info.tcb_level.tcb_status.fields.outofdate == 1);
+    printf("\n\nQE Identity V2 positive test, OutOfDate. PASSED.\n");
+
+    // QE Identity V2 positive with advisoryIDs
+    platform_tcb_level.isvsvn[0] = 2;
+    OE_TEST(
+        test_verify_qe_identity_info(
+            enclave,
+            &ecall_result,
+            (const char*)&positive_qe_id_info_with_advisoryids[0],
+            &platform_tcb_level,
+            &parsed_info) == OE_OK);
+    OE_TEST(ecall_result == OE_OK);
+    OE_TEST(parsed_info.id == QE_IDENTITY_ID_QE);
+    OE_TEST(parsed_info.tcb_level.tcb_status.fields.up_to_date == 1);
+    OE_TEST(parsed_info.tcb_level.advisory_ids_size > 0);
+    OE_TEST(
+        parsed_info.tcb_level.advisory_ids_offset <
+        positive_qe_id_info_with_advisoryids.size());
+
+    const char* ptr = (const char*)&positive_qe_id_info_with_advisoryids
+        [parsed_info.tcb_level.advisory_ids_offset];
+    OE_TEST(
+        oe_parse_advisoryids_json(
+            (const uint8_t*)ptr,
+            parsed_info.tcb_level.advisory_ids_size,
+            (const uint8_t**)&advisoryIDs,
+            2,
+            (size_t*)&advisoryIDs_length,
+            2,
+            &num_advisory_ids) == OE_OK);
+    OE_TEST(num_advisory_ids == 2);
+    for (int i = 0; i < 2; i++)
+    {
+        printf(
+            "AdvisoryIDs[%d]: %.*s\n",
+            i,
+            (int)advisoryIDs_length[i],
+            advisoryIDs[i]);
+        OE_TEST(
+            strncmp(
+                (const char*)advisoryIDs[i],
+                expectedAdvisoryIDs[i],
+                advisoryIDs_length[i]) == 0);
+    }
+    printf("QE Identity V2 positive test, with advisoryIDs. PASSED\n");
+
+    // QVE Identity V2 positive test
+    platform_tcb_level.isvsvn[0] = 2;
+    OE_TEST(
+        test_verify_qe_identity_info(
+            enclave,
+            &ecall_result,
+            (const char*)&positive_qve_id_info[0],
+            &platform_tcb_level,
+            &parsed_info) == OE_OK);
+    OE_TEST(ecall_result == OE_OK);
+    OE_TEST(parsed_info.id == QE_IDENTITY_ID_QVE);
+    check_parsed_v2_values(parsed_info);
+    printf("\n\nQVE Identity V2 positive test. PASSED.\n");
+
+    // negative test without a valid platform_tcb_level
+    OE_TEST(
+        test_verify_qe_identity_info(
+            enclave,
+            &ecall_result,
+            (const char*)&positive_qe_id_info[0],
+            NULL,
+            &parsed_info) == OE_OK);
+    OE_TEST(ecall_result == OE_INVALID_PARAMETER);
+    printf("\n\nQE Identity V2 negative test with invalid platform_tcb_level. "
```

Comment:
This file begs the question... why do we need a custom test framework here rather than using an existing widely used test framework like gtest?  (In contrast, I believe the feature.new_platforms branch uses gtest)

Reply:
I am open to other test frameworks, this follows the existing established test framework/pattern.  Would you mind sharing the design doc/proposal?  As for this PR, I suggest creating a issue to update the tests to use the alternative test framework once agreed.

Reply:
Filed issue #2350